### ISSUE TITLE
Issues with Test Guidance for the Hill Cipher

### ISSUE THREAD

**Issue by jlwether** [toebes/ciphers#124]
The test guidance for the Hill Cipher states that "In general, [the matrix] is more likely to be invertible if you use the letters B, D, F, H, L, N, R, T, X and Z. as they are odd and non-prime, but you can mix in some other letters."  The explanation "as they are odd and non-prime" is misleading.  

The determinant will be invertible (mod 26) if and only if it is invertible modulo both 2 and 13.  **All other primes are completely irrelevant!**  Moreover, 13 is large enough that we can probably afford to not worry about it; a (roughly) 1/13 chance of failure seems acceptable to me.  However, the even/odd question can be annoying, especially since all vowels are even.  

For a 2x2 matrix we can be explicit.  Since it is very likely that we will want either the 2nd or 3rd letter to be a vowel, I think the more useful advice would be that one or both of the 2nd and 3rd letters should be vowels, while both the first and last letters come from the set of odd letters: B, D, F, H, J, L, N, P, R, T, V, X,  and Z.  This would guarantee that the determinant is odd, and then we only need to check to make sure it isn't 0 (mod 13), which means we would succeed with probability 12/13.

For a 3x3 matrix it isn't as easy, but if we ask for about half the letters to be odd (same set as before), then there is a good chance that the determinant will be odd.  In fact, assuming random placement of the odd letters, the exact odds for each possible number of odd letters is as follows:

- 0: 0.000
- 1: 0.000
- 2: 0.000
- 3: 0.071
- 4: 0.286
- 5: 0.571
- 6: 0.426
- 7: 0.500
- 8: 0.000
- 9: 0.000

Basically, we want 5, 6, or 7 odd letters in the mix in order to have a decent shot at being invertible mod 2. If we are lucky, we may get by with 4 odd letters; with only 3 odd letters we need to be **extremely** lucky; and for any other number of odd letters, it is simply impossible.

Of course, this is probably too much analysis for the test guidance.  For that, my recommendation is to simply say that "for both 2x2 and 3x3 matrices, we can improve the likelihood that the determinant is invertible if half to three-quarters of the letters are odd; the odd letters are B, D, F, H, J, L, N, P, R, T, V, X,  and Z."

**Comment by jlwether** [comment]
The test guidance once again says, "In general, it is more likely to be invertible if you use the letters B, D, F, H, L, N, R, T, X and Z. as they are odd and non-prime, but you can mix in some other letters."  I have the same comment on this as before, though I suspect this time it was just an old version that accidentally got copied into the current text.
